### DISCUSSION TITLE
How to call a separate script from Argo workflow Scripts ?

### DISCUSSION THREAD

**Discussion by pranavdi007** [argoproj/argo-workflows#8787] [Q&A] [answered]
I have a use case where i have to write some common code in all the argo workflow scripts .
The use case is i have to perform some common action in workflow steps so doing the same task will have common code in that case in every script. So I thought of putting that part of code in another script and calling that from my main step . But I am not able to access that script from inside the main script. What is the best way to do this ? I had put the script in the same folder where other workflow scripts were present.

**Comment by terrytangyuan** [comment, marked as answer]
You can add init containers to your steps: https://github.com/argoproj/argo-workflows/blob/master/examples/init-container.yaml
